smac/facade/roar_facade.py

```diff
@@ -0,0 +1,130 @@
+import logging
+
+import numpy as np
+
+from smac.tae.execute_ta_run import ExecuteTARun
+from smac.tae.execute_ta_run import StatusType
+from smac.stats.stats import Stats
+from smac.scenario.scenario import Scenario
+from smac.runhistory.runhistory import RunHistory
+from smac.runhistory.runhistory2epm import AbstractRunHistory2EPM, RunHistory2EPM4LogCost, RunHistory2EPM4Cost
+from smac.initial_design.initial_design import InitialDesign
+from smac.intensification.intensification import Intensifier
+from smac.smbo.acquisition import AbstractAcquisitionFunction
+from smac.epm.random_epm import RandomEpm
+from smac.facade.smac_facade import SMAC
+
+__author__ = "Marius Lindauer"
+__copyright__ = "Copyright 2016, ML4AAD"
+__license__ = "3-clause BSD"
+
+
+class ROAR(SMAC):
+
+    def __init__(self,
+                 scenario: Scenario,
+                 tae_runner: ExecuteTARun=None,
+                 runhistory: RunHistory=None,
+                 intensifier: Intensifier=None,
+                 acquisition_function: AbstractAcquisitionFunction=None,
+                 model=None,
+                 runhistory2epm: AbstractRunHistory2EPM=None,
+                 initial_design: InitialDesign=None,
+                 stats: Stats=None,
+                 rng: np.random.RandomState=None):
+        '''
+        Facade to use ROAR default mode
+
+        Parameters
+        ----------
+        scenario: smac.scenario.scenario.Scenario
+            Scenario object
+        tae_runner: ExecuteTARun
+            object that implements the following method to call the target
+            algorithm (or any other arbitrary function):
+            run(self, config)
+            If not set, it will be initialized with the tae.ExecuteTARunOld()
+        runhistory: RunHistory
+            runhistory to store all algorithm runs
+        intensifier: Intensifier
+            intensification object to issue a racing to decide the current incumbent
+        acquisition_function : AcquisitionFunction
+            Object that implements the AbstractAcquisitionFunction. Will use
+            EI if not set.
+        model : object
+            Model that implements train() and predict(). Will use a
+            RandomForest if not set.
+        runhistory2epm : RunHistory2EMP
+            Object that implements the AbstractRunHistory2EPM. If None,
+            will use RunHistory2EPM4Cost if objective is cost or
+            RunHistory2EPM4LogCost if objective is runtime.
+        initial_design: InitialDesign
+            initial sampling design
+        stats: Stats
+            optional stats object
+        rng: np.random.RandomState
+            Random number generator
+        '''
+        self.logger = logging.getLogger("ROAR")
+
+        # initialize random number generator
```

Comment:
This code duplication hints that setting the rng should be a private method in the SMAC class which can then be called in both constructors.